app/models/mixins/yaml_import_export_mixin.rb

```diff
@@ -37,6 +38,10 @@ def import(fd, options = {})
         raise "Invalid YAML file"
       end
 
+      if options.key?(:expected_class) && (options[:expected_class] != reps[0].keys.first)
+        raise _("Incorrect format: Expected #{options[:expected_class]} and received #{reps[0].keys.first}")
+      end
```

Comment:
@bmclaughlin I'm ok with this check, but it looks like doing this for `MiqReport` (ManageIQ/manageiq-ui-classic#2227) would negate the need for the `rescue` below on line 59: `# for the legacy MiqReport` as we would never get to that code.

@gtanzillo Can you comment on this or recommend someone?